### ISSUE TITLE
Subclass models not working

### ISSUE THREAD

**Issue by owensbla** [mperham/sidekiq#2202]
We have various stat subclasses based off a parent. For example:

`BasketballStat::Offensive::Event`, `BasketballStat::Offensive::Average`, `BasketballStat::Offensive::Total` all subclass from `BasketballStat::Offensive`.

Now, let's say we spin up a worker and run a task that fetches basketball stats from a third party, update, and save them (::Event stats). We have an after_save signal on BasketballStat::Offensive that checks if the subclass type being saved is ::Event and calls a method that updates ::Average and ::Total rows.

This is where things start to get weird. In this method, we can use binding.remote_pry, run `BasketballStat::Offensive::Average.first`, and we're returned an ::Event subclass. In fact, regardless of the subclass we use (::Average, ::Total), it returns an ::Event subclass.

I thought it might be eager loading related, but I'm at a total loss. Does anyone have any thoughts?

Note, when trying to fetch a stat subclass through the following technique (where basketball_offensive_stat_type is the class' inheritance_column):

```
BasketballStat::Offensive.where(:basketball_offensive_stat_type => "BasketballStat::Offensive::Average")
```

We get back an empty array. Which is not accurate.

Another note, when trying any of the above in when using `rails console` it works as expected.

**Comment by mperham** [comment]
Any news?

**Comment by owensbla** [comment]
Nope. We ended up scheduling the task on it's own worker to resolve the issue.

To clarify a bit more:

We basically wanted each player to update their average and total stats each time a event or season stat changed (which are all BasketballStat subclasses). So, we had a method on BasketballStat that was called on after_save when the subclass stat type was ::Event or ::Season and then we called the corresponding player's update_stats method when that was the case. With this setup, the above issue was occurring when fetching and updating ::Event and ::Season stats and saving them (thus triggering BasketballPlayer#update_stats through the BasketballStat.player relation).

So, for us, the fix was to just schedule a daily task to loop through each BasketballPlayer and recalculate their ::Average and ::Total stats. When calling BasketballPlayer#update_stats this way, there were no issues with subclasses.
